c2cgeoportal/views/entry.py

```diff
@@ -900,6 +898,13 @@ def _wms_layers(self, role_id, ogc_server):
 
         return wms, list(wms.contents), wms_errors
 
+    @cache_region.cache_on_arguments()
+    def _load_tree_itmes(self, cache_version):
+        self._themes_cache = DBSession.query(Theme).options(joinedload(Theme.functionalities)).all()
+        self._layerswms_cache = DBSession.query(LayerWMS).options(joinedload(LayerWMS.dimensions)).all()
+        self._layerswmts_cache = DBSession.query(LayerWMTS).options(joinedload(LayerWMTS.dimensions)).all()
+        self._treeitem_cache = DBSession.query(with_polymorphic(TreeItem, [LayerGroup, LayerV1])).all()
+
```

Comment:
Pretty sure you could load other things here like children_relation, ui_metadata, it should be possible to render themes in less that 1s without cache. You should take a look at sqlalchemy logs in debug mode. Concerning dimensions you should use a subqueryload (do not return all LayerWMS fields for each dimension value).

Reply:
You could have at least only one request per table

Reply:
I can do some tests on `children_relation`, 
I will add `ui_metadata`

> it should be possible to render themes in less that 1s without cache

with WMS capabilities requests that take more than 8s. :-/

I will do a quick test on `subqueryload`.

Thanks for your feedback @arnaud-morvan :-)

Reply:
joinedload should be used when you have only one remote record (ogc_server for example) in case your relation is a collection (uselist = True), it should be better to use subqueryload

Reply:
this will populate all children relations in one database request: 
```
groups = self._dbsession.query(TreeGroup). \
            options(subqueryload('children_relation')). \
all()
```